config/confighttp/confighttp.go

```diff
@@ -107,6 +109,11 @@ type HTTPServerSettings struct {
 	// An empty list means that CORS is not enabled at all. A wildcard (*) can be
 	// used to match any origin or one or more characters of an origin.
 	CorsOrigins []string `mapstructure:"cors_allowed_origins"`
+
+	// EnableDecompression if true, a middleware is configured that decompresses the body
+	// of incoming HTTP requests based on the compression format in the Content-Encoding header.
+	// Currently, it has support for gzip and deflate/zlib.
+	EnableDecompression bool `mapstructure:"enable_decompression"`
```

Comment:
I expect we need a mechanism to enable/disable the middleware so added this. If we don't we can remove this...
The naming is tricky because `enable_decompression` is an internal detail that may not be understandable by the end-user. 
Will a`Compression` setting with a comment stating only request compression is handled work?

Reply:
Realized `EnableDecompression ` is ambiguous too it could also mean `response` decompression. :(. 

Would something like `OffloadCompression` be an appropriate name?